remo/dashboard/views.py

```diff
@@ -170,8 +170,15 @@ def dashboard(request):
         args['reps_without_profile'] = reps.filter(userprofile__registration_complete=False)
 
     if user.groups.filter(Q(name='Admin') | Q(name='Council') | Q(name='Peers')).exists():
+        # We want to show
+        #   - Reps with completed profiles and no mentor
+        #   - Reps with mentors who are Alumni
+        # while not including any Alumni Reps
         args['reps_without_mentors'] = reps.filter(
-            userprofile__registration_complete=True, userprofile__mentor=None)
+            ~Q(groups__name='Alumni') &
+            ((Q(userprofile__registration_complete=True) & Q(userprofile__mentor=None)) |
+                Q(userprofile__mentor__groups__name='Alumni'))
```

Comment:
Maybe you could structure your query to something like:

```python
alumni_ids = User.objects.filter(groups__name='Alumni')
reps_with_alumni_mentors_q = Q(userprofile__mentor__id__in=alumni_ids)
reps_without_mentor_q = Q(userprofile__registration_complete=True, userprofile__mentor__isnull=True)
reps_without_mentors = User.objects.filter(reps_with_alumni_mentors_q | reps_without_mentor_q)
```

Reply:
Fyi, I haven't spent much time testing this

Reply:
This is way more readable, will do. Only problem I see with this is that we still need to filter on the reps, not all users (see above).